dashboard/app/models/teacher_feedback.rb

```diff
@@ -26,12 +26,20 @@
 
 class TeacherFeedback < ApplicationRecord
   acts_as_paranoid # use deleted_at column instead of deleting rows
-  validates_presence_of :student_id, :level_id, :teacher_id, :script_level_id, unless: :deleted?
+  validates_presence_of :student_id, :script_id, :level_id, :teacher_id, :script_level_id, unless: :deleted?
   belongs_to :student, class_name: 'User'
   has_many :student_sections, class_name: 'Section', through: :student, source: 'sections_as_student'
+  belongs_to :script
   belongs_to :level
   belongs_to :script_level
   belongs_to :teacher, class_name: 'User'
+  validate :validate_script_and_script_level, on: :create
```

Comment:
only do this on create, because we want update to succeed at https://github.com/code-dot-org/code-dot-org/blob/4b0b0c00a084714dfcfd14a18a6eab2414eead91/dashboard/app/models/teacher_feedback.rb#L83 even if the script_level_id has become invalid.